### PR TITLE
Shortcuts help modal

### LINKED ISSUE #530: Document keyboard shortcuts
**Is your feature request related to a problem? Please describe.**
Just lost my position in a lecture recording as a result of using the cmd+1 keyboard shortcut (i'm on a mac, so i guess the linux/windows equivalents would be different).
The shortcuts `cmd+n` (with $n \in \\{1, 2, ..., 8\\}$) in Chrome will usually switch to the n-th tab from the left, but it seems like the video player overrides these commands to jump to different sections of a video. However, this is not documented anywhere.

**Describe the solution you'd like**
Document the keyboard commands.
Many web-based applications (such as e.g. Gmail or GitHub) are using the shift+? keyboard shortcut to present an overlay/menu listing all possible keyboard shortcuts, so this might be a good thing to do as well.
Documenting the commands would:
- make it clearer what they do exactly (i'm still not sure what exactly the overwritten cmd+n commands do. do they always jump in 14-minute intervals, or do they jump to 1/10-of-the-whole-video timestamps?)
- make the commands more discoverable. (after realising that the cmd+n commands exist i continued to press some random buttons, and it seems like `j`, `k`, `l`, `m`, `z`, and `x` also are mapped to certain actions, but then again it is impossible to easily know what they do exactly, without delving into the source code of the website)

**Describe alternatives you've considered**
- Just not do anything and leave it as-is
- Add a section listing all available keyboard shortcuts somewhere else (e.g. in the settings page)
- Have some kind of questionmark-button at the bottom right of the video player, next to the camera mode and feedback buttons

### PR DESCRIPTION
**Pull request by MatthiasReumann**
## Changes 
- Closes #530 
- Open help modal on `SHIFT+?`
- Close modal on outside click

## Screenshots
image

### CODE REVIEW

File: web/template/watch.gohtml [hunk 1]
```diff
@@ -16,11 +16,19 @@
...
       x-on:pauseend.window="paused=false"
+      @keypress.shift.window="(e) => watch.onShift(e)"
       x-init="new watch.Watch(); watch.startWebsocket(); seekLogger.attach();">
 {{template "header" .IndexData.TUMLiveContext}}
+<!-- Keyboard shortcuts modal -->
+<div id = "shortcuts-help-modal" class = "hidden flex absolute top-0 h-screen w-screen z-50 backdrop-blur-sm">
+    <div class = "m-auto" @click.outside="watch.toggleShortcutsModal();">
+        {{template "shortcuts"}}
+    </div>
+</div>
```
Comment (joschahenningsen) [inline web/template/watch.gohtml]:
```suggestion
<div id="shortcuts-help-modal" class="hidden flex absolute top-0 h-screen w-screen z-50 backdrop-blur-sm">
    <div class="m-auto" @click.outside="watch.toggleShortcutsModal();">
        {{template "shortcuts"}}
    </div>
</div>
```

File: web/template/partial/shortcuts.gohtml [hunk 2]
```diff
@@ -0,0 +1,87 @@
+{{define "shortcuts"}}
+    <div class="border rounded-lg shadow text-3 bg-white dark:bg-secondary dark:border-gray-800 dark:shadow-0">
+        <div class = "border-b dark:border-gray-800 px-5 py-1">
+            <h6 class = "text-sm font-bold">Shortcuts</h6>
```
Comment (joschahenningsen) [inline web/template/partial/shortcuts.gohtml]:
```suggestion
            <h6 class="text-sm font-bold">Shortcuts</h6>
```

File: web/template/partial/shortcuts.gohtml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        </div>
+        <div class = "grid gap-2 pb-3 pt-2 px-5">
```
Comment (joschahenningsen) [inline web/template/partial/shortcuts.gohtml]:
```suggestion
        <div class="grid gap-2 pb-3 pt-2 px-5">
```

File: web/template/partial/shortcuts.gohtml [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+                <p class="text-sm align-middle m-0">Increase volume</p>
+            </div>
+            <div class="flex justify-between">
+                <div class="flex bg-gray-100 dark:bg-secondary-light text-5 px-2 py-1 rounded mr-3">
+                    <i class="text-xs fa-solid fa-arrow-down my-auto"></i>
+                </div>
+                <p class="text-sm align-middle m-0">Decrease volume</p>
+            </div>
+            <div class="flex justify-between">
+                <div class = "flex">
```
Comment (joschahenningsen) [inline web/template/partial/shortcuts.gohtml]:
```suggestion
                <div class="flex">
```

File: web/template/partial/shortcuts.gohtml [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+                    <div class="flex bg-gray-100 dark:bg-secondary-light text-5 px-2 py-1 rounded">
+                        <span class="text-xs font-semibold my-auto">1</span>
+                    </div>
+                    <span class = "font-semibold my-auto mx-2">-</span>
```
Comment (joschahenningsen) [inline web/template/partial/shortcuts.gohtml]:
```suggestion
                    <span class="font-semibold my-auto mx-2">-</span>
```

File: web/template/partial/shortcuts.gohtml [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+            </div>
+            <div class="flex justify-between">
+                <div class="flex bg-gray-100 dark:bg-secondary-light text-5 px-2 py-1 rounded">
+                    <span class="text-xs font-semibold my-auto">0</span>
+                </div>
+                <p class="text-sm align-middle m-0">Seek to the beginning of the video</p>
+            </div>
+            <div class="flex justify-between">
+                <div class="flex bg-gray-100 dark:bg-secondary-light text-5 px-2 py-1 rounded">
+                    <span class="text-xs font-semibold my-auto"> < </span>
```
Comment (joschahenningsen) [inline web/template/partial/shortcuts.gohtml]: Would this be safer or does this make no difference?
```suggestion
                    <span class="text-xs font-semibold my-auto"> &lt; </span>
```

**Review by joschahenningsen — CHANGES_REQUESTED** [review]
Looks great, thanks! For consistency we should remove spaces in our html according to this styleguide: https://www.w3schools.com/html/html5_syntax.asp

File: web/template/partial/shortcuts.gohtml (hunk 6 above)
Comment (MatthiasReumann) [reply web/template/partial/shortcuts.gohtml]: Yup. That's definitely an improvement and less confusing.

**Review by joschahenningsen — APPROVED** [review]
Thanks!

**Comment by just-max** [comment]
Looks nice, thanks!

**Pull request merged**